_posts/2021-01-25-solidity-developer-survey-2020-results.md

```diff
@@ -0,0 +1,358 @@
+
+### (4) Most Liked, Dreaded and Anticipated Features
+
+We've asked survey participants what they like and dislike most about Solidity and what they would like to see the most in upcoming Solidity versions. Those questions had free-text reply fields so we've summarized the replies and are listing the most mentioned features and aspects below.
+
+#### Most Liked Features
+
+Simplicity and cleanliness has been the number one aspect that the participants value about Solidity. 
+
+Furthermore, the developers appreciated the improvements in available tooling, especially highlighting Remix and Hardhat several times. 
+
+Another aspect the respondents appreciate is that Solidity is "easy to learn" and "simple to code".
+
+Several people stated that Solidity is the right tool for what they are working on and that they like the focus on smart contracts and Solidity being a "special purpose tool" for it.
+
+Other features and aspects that have been mentioned repeatedly as cherished and appreciated are:
+- Mappings
+- Human readability
+- Structs
+- Modifiers
+- Strong-typing
+- JS-likeness
+- Assembly access
+- SafeMath by default
+- Good library support
+- C++-likeness
+- SMTChecker
+- Composability
+- Immutability
+- ABI Encoder V2
+- Explicitness
+- Inheritance
+- Compact code size
+- Syntax
+- Steady improvement of the language
+
+
+#### Most Dreaded or Disliked Features
+
+The currently most dreaded aspect about Solidity is clearly debugging, followed by a lack of Solidity libraries. 
+
+Respondents also complain about inheritance conflicts and the need to write more specific code in assembly / low-level code. 
+
+Other aspects and features that have been mentioned repeatedly as dreaded or disliked are:
+- Error handling
+- Memory management
+- Frequency of breaking changes and dependencies breaking
+- Contract & stack size limits
+- Inconsistency with other languages
+- ABI encode
+- Unclear gas usage / gas inefficiency
+- Tooling
+- Complexity
+- Recompiles failing due to metadata
+- Verbosity
+- Dynamic array handling
+- Testing
+
+#### Most Anticipated Features
+
+The most anticipated feature by far was "no more SafeMath". SafeMath by default has been [introduced with 0.8.0](https://blog.soliditylang.org/2020/12/16/solidity-v0.8.0-release-announcement/) and hence is not an issue anymore. 🥳
+
+The second most anticipated feature was better debugging and more debugging tools. 
+
+Also ranked high was better documentation, especially with a focus on more tutorials and code examples.
+
+Other anticipated and frequently mentioned features were:
+- Gas efficiency optimization features
+- Generics
+- console.log()
+- Dynamic arrays
+- Better tooling
+- Better IDEs
+- LSP support
+- Structs
+- Rich revert reasons/ error codes
+- Standard library
+- Custom optimization features
+- Release v.1.0.0
+- SMTChecker
+- Stable ABIEncoderV2
+- No more stack too deep
+- Decimals
+- Over- and underflow handling
+- Fixed point types
```

Comment:
I'd group decimals together with fixed-point types. Maybe people even meant the same thing by it. Technically fixed-point numbers stored as decimal numbers (rather than binary ones) would be a separate feature but I suspect that people were just casually referring to fractional numbers as decimals.